lib/sqlalchemy/sql/base.py

```diff
@@ -1882,6 +1886,108 @@ def __hash__(self):
         return hash(tuple(x for x in self))
 
 
+_KT = TypeVar("_KT", bound=str)
+_VT = TypeVar("_VT")
+_VT_co = TypeVar("_VT_co", covariant=True)
+
+
+class DictCollection(_collections_abc.MutableMapping, Generic[_KT, _VT]):
```

Comment:
we already have this, it's Properties at https://github.com/sqlalchemy/sqlalchemy/blob/b51a5f596299e1e52ca27b2db20394110f584257/lib/sqlalchemy/util/_collections.py#L170 .  let's remove this and use that (we have enough collection classes...)

Reply:
Wow, perfect. Guess I completely reinvented a good wheel